lib/ansible/modules/system/nosh.py

```diff
@@ -0,0 +1,381 @@
+        # get status information
+        result['status'] = get_service_status(module, service_path)
+        running = service_is_running(result['status'])
+
+        if module.params['state'] == 'started':
+            if not running:
+                action = 'start'
+        elif module.params['state'] == 'stopped':
+            if running:
+                action = 'stop'
+        # reset = start/stop according to enabled status
+        elif module.params['state'] == 'reset':
+            if enabled is not running:
+                if running:
+                    action = 'stop'
+                    result['state'] = 'stopped'
+                else:
+                    action = 'start'
+                    result['state'] = 'started'
+        # start if not running, 'service' module constraint
+        elif module.params['state'] == 'restarted':
+            if not running:
+                action = 'start'
+                result['state'] = 'started'
+            else:
+                action = 'condrestart'
+        # start if not running, 'service' module constraint
+        elif module.params['state'] == 'reloaded':
+            if not running:
+                action = 'start'
+                result['state'] = 'started'
+            else:
+                action = 'hangup'
+
+    # change state as needed
+    if action:
+        result['changed'] = True
+        if not module.check_mode:
+            (rc, out, err) = run_sys_ctl(module, [action, service_path])
+            if rc != 0:
+                module.fail_json(msg="Unable to %s service %s: %s" % (action, service_path, err))
+
+# ===========================================
+# Main control flow
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            state=dict(choices=['started', 'stopped', 'reset', 'restarted', 'reloaded'], type='str'),
+            enabled=dict(type='bool'),
+            preset=dict(type='bool'),
+            user=dict(type='bool'),
+        ),
+        supports_check_mode=True,
+        mutually_exclusive=[['enabled', 'preset']],
+    )
+
+    service = module.params['name']
+    rc = 0
+    out = err = ''
+    result = {
+        'name': service,
+        'changed': False,
+        'status': {},
+    }
+
+    # check service can be found (or fail) and get path
+    service_path = get_service_path(module, service)
+    result['service_path'] = service_path
+
+    # set enabled state, service need not be loaded
+    if module.params['enabled'] is not None or module.params['preset']:
+        handle_enabled(module, result, service_path)
+
+    # set service running state
+    if module.params['state'] is not None:
+        handle_state(module, result, service_path)
+
+    # get final service status if possible
+    if service_is_loaded(module, service_path):
+        result['status'] = get_service_status(module, service_path)
+    else:
+        result['status'] = {'Loaded': False}
```

Comment:
I think there's two consistent ways to do it here:

1. Add the 'Loaded' key whether full status is obtained or not. This is more explicit but artificially extends the original API.
One can then test for full status information with `if result['status']['Loaded']: ...`

2. Return an empty dictionary (or `None`?) in the non-loaded case. Possibly less clear but just as simple to test (`if result['status']: ...`)

I think 2. is probably better. It makes the 'status' dictionary more like an optional value, which is the idea.